text/0000-trait-alias.md

```diff
@@ -0,0 +1,270 @@
+- Feature Name: Trait alias
+- Start Date: 2016-08-31
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Traits can be aliased with the `trait TraitAlias = …;` construct. Currently, the right hand side is
+a bound – a single trait, a combination with `+` traits and lifetimes. Type parameters and
+lifetimes can be added to the *trait alias* if needed.
+
+# Motivation
+[motivation]: #motivation
+
+## First motivation: `impl`
+
+Sometimes, some traits are defined with parameters. For instance:
+
+```rust
+pub trait Foo<T> {
+  // ...
+}
+```
+
+It’s not uncommon to do that in *generic* crates and implement them in *backend* crates, where the
+`T` template parameter gets substituted with a *backend* type.
+
+```rust
+// in the backend crate
+pub struct Backend;
+
+impl trait Foo<Backend> for i32 {
```

Comment:
This is a typo, right? That is, It should just be `impl Foo<Backend> for i32`, not `impl trait Foo`, right?